packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -25,4 +41,63 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         });
     }
 
+    @postConstruct()
+    protected async init() {
+        this.navigatorModel = (await this.widget).model;
+
+        await this.fileNavigatorPreferences.ready;
+        this.shell.currentChanged.connect(() => this.onCurrentWidgetChangedHandler());
+    }
+
+    registerCommands(registry: CommandRegistry): void {
+        super.registerCommands(registry);
+        registry.registerCommand(FileNavigatorCommands.REVEAL_IN_NAVIGATOR, {
+            execute: () => this.openView({ activate: true }).then(() => this.selectWidgetFileNode(this.shell.currentWidget)),
+            isEnabled: () => Navigatable.is(this.shell.currentWidget),
+            isVisible: () => Navigatable.is(this.shell.currentWidget)
+        });
+    }
+
+    registerMenus(registry: MenuModelRegistry): void {
+        super.registerMenus(registry);
+        registry.registerMenuAction(SHELL_TABBAR_CONTEXT_MENU, {
+            commandId: FileNavigatorCommands.REVEAL_IN_NAVIGATOR.id,
+            label: 'Reveal in Files',
+            order: '5'
+        });
+    }
+
+    registerKeybindings(registry: KeybindingRegistry): void {
+        super.registerKeybindings(registry);
+        registry.registerKeybinding({
+            command: FileNavigatorCommands.REVEAL_IN_NAVIGATOR.id,
+            keybinding: "alt+r"
+        });
+    }
+
+    /**
+     * Reveals and selects node in the file navigator to which given widget is related.
+     * Does nothing if given widget undefined or doesn't have related resource.
+     *
+     * @param widget widget file resource of which should be revealed and selected
+     */
+    async selectWidgetFileNode(widget: Widget | undefined): Promise<void> {
+        if (Navigatable.is(widget)) {
+            const fileUri = widget.targetUri;
+            if (fileUri) {
+                this.navigatorModel.revealFile(fileUri).then((node: ITreeNode) => {
```

Comment:
you can use await here to get the model and revean a file, it looks more readable and does not require a new property:
```ts
const { model } = await this.widget;
const node = model.revealFile(fileUri);
if (ISelectableTreeNode.is(node)) {
    model.selectNode(node);
}
```

Reply:
Thanks